Replace PLINY_ENV with APP_ENV

soulim opened the pull request:
Resolves https://github.com/interagent/pliny/issues/277.

`PLINY_ENV` and `pliny_env` have been replaced with `APP_ENV` and `app_env` everywhere in the code.

In order to help with transition, a deprecation message is displayed if `Config.pliny_env` is used. So `pliny_env` is actually deprecated and can be still called with a grumpy message. Deprecation message is based on [the pattern used in Sinatra](https://github.com/sinatra/sinatra/blob/1b0edc0aeaaf4839cadfcec1b21da86e6af1d4c0/lib/sinatra/base.rb#L945).

Would be happy to hear your thoughts on this one 😄

brandur commented on lib/pliny/config_helpers.rb:
```diff
@@ -50,12 +50,20 @@ def array(method = nil)
 
     def rack_env
-      if pliny_env == "development" || pliny_env == "test"
+      if app_env == "development" || app_env == "test"
```
I wonder if we should respect the value of `app_env` but _also_ `pliny_env` here for a bit longer to give people some opportunity to upgrade?

I'm a little worried that with this if someone bumped their Pliny version behavior based off `rack_env` would change without warning in their environments because they hadn't injected the new `APP_ENV` variable into `.env` and/or their app's config.

Thoughts?

soulim replied:
Yeah... Now when you said that, I also think backward compatibility with `pliny_env` could be better. I'll improve the solution and will add more tests on cases when `PLINY_ENV` is still defined in the environment.

Thanks for the idea! 👍

brandur commented:
Excellent! Thank you :)

On Tuesday, October 25, 2016, Alexander Sulim <notifications@github.com>
wrote:

> *@soulim* commented on this pull request.
> ------------------------------
>
> In lib/pliny/config_helpers.rb
> <https://github.com/interagent/pliny/pull/282>:
>
> > @@ -50,12 +50,20 @@ def array(method = nil)
>      end
>
>      def rack_env
> -      if pliny_env == "development" || pliny_env == "test"
> +      if app_env == "development" || app_env == "test"
>
> Yeah... Now when you said that, I also think backward compatibility with
> pliny_env could be better. I'll improve the solution and will add more
> tests on cases when PLINY_ENV is still defined in the environment.
>
> Thanks for the idea! 👍
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/interagent/pliny/pull/282>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAF6erR0BJOUJUEhQ1h1m3kCqWhxKxsjks5q3bgygaJpZM4KbINP>
> .
>

soulim commented:
Updated solution according to our discussion 😄  In case `PLINY_ENV` is still around, it will be used, but with appropriate warnings.

Additionally I've added short comments to new methods. I hope they would be helpful for other developers.

What do you think?

brandur commented:
> Additionally I've added short comments to new methods. I hope they would be helpful for other developers.

Love it. Thanks!

> What do you think?

Looks good to me. Pulling this in.
